WebViewControl/WebView.JavascriptExecutor.cs

```diff
@@ -42,185 +42,180 @@ internal class ScriptTask {
-                            OwnerWebView.ForwardUnhandledAsyncException(ParseException(e, evaluatedScriptFunctions), frameName);
+                try {
+                    var scriptsToExecute = new List<ScriptTask>();
+                    foreach (var scriptTask in PendingScripts.GetConsumingEnumerable()) {
+                        if (scriptTask.Evaluate == null) {
+                            scriptsToExecute.Add(scriptTask);
                         }
+                        if ((PendingScripts.Count == 0 || scriptTask.Evaluate != null) && scriptsToExecute.Count > 0) {
+                            BulkExecuteScripts(scriptsToExecute);
+                            scriptsToExecute.Clear();
+                        }
+                        scriptTask.Evaluate?.Invoke(scriptTask.Script);
                     }
+                } catch (OperationCanceledException) {
+                    // stop
+                } finally {
+                    PendingScripts.Dispose();
+                    FlushTaskCancelationToken.Dispose();
                 }
+            }
 
-                if (scriptToEvaluate != null) {
-                    scriptToEvaluate.Evaluate();
+            private void BulkExecuteScripts(IEnumerable<ScriptTask> scriptsToExecute) {
+                var script = string.Join(";" + Environment.NewLine, scriptsToExecute.Select(s => s.Script));
+                if (frame.IsValid) {
+                    var frameName = frame.Name;
+                    try {
+                        var timeout = OwnerWebView.DefaultScriptsExecutionTimeout ?? DefaultTimeout;
+                        var task = OwnerWebView.chromium.EvaluateJavaScript<object>(WrapScriptWithErrorHandling(script), timeout: timeout);
+                        task.Wait(FlushTaskCancelationToken.Token);
+                    } catch (OperationCanceledException) { 
+                        // ignore
+                    } catch (Exception e) {
+                        var evaluatedScriptFunctions = scriptsToExecute.Select(s => s.FunctionName);
+                        OwnerWebView.ForwardUnhandledAsyncException(ParseException(e, evaluatedScriptFunctions), frameName);
+                    }
                 }
             }
 
-            public T EvaluateScript<T>(string script, string functionName = null, TimeSpan? timeout = null) {
-                var result = default(T);
-                Exception exception = null;
-                var waitHandle = new ManualResetEvent(false);
-                var scriptWithErrorHandling = WrapScriptWithErrorHandling(script);
-
-                void Evaluate() {
-                    var effectiveTimeout = timeout ?? OwnerWebView.DefaultScriptsExecutionTimeout ?? DefaultTimeout;
+            public async Task<T> EvaluateScript<T>(string script, string functionName = null, TimeSpan? timeout = null) {
+                const string TimeoutExceptionName = "Timeout";
```

Comment:
We can declare this at class level

Reply:
its only use in this method... prefer this way.. stays in context